SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -94,20 +96,20 @@ void ASpatialFunctionalTest::OnAuthorityGained()
 
 void ASpatialFunctionalTest::RegisterAutoDestroyActor(AActor* ActorToAutoDestroy)
 {
-	if (HasAuthority())
+	if (ActorToAutoDestroy == nullptr)
 	{
-		Super::RegisterAutoDestroyActor(ActorToAutoDestroy);
+		UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("Actor to auto destroy is NULL"));
 	}
-	else if(LocalFlowController != nullptr)
+	else if (ActorToAutoDestroy->HasAuthority())
 	{
-		if(LocalFlowController->ControllerType == ESpatialFunctionalTestFlowControllerType::Server)
-		{
-			CrossServerRegisterAutoDestroyActor(ActorToAutoDestroy);
-		}
-		else
-		{
-			ServerRegisterAutoDestroyActor(ActorToAutoDestroy);
-		}
+		// Add component to actor to auto destroy when test finishes
+		USpatialFunctionalTestAutoDestroyComponent* AutoDestroyComponent = NewObject<USpatialFunctionalTestAutoDestroyComponent>(ActorToAutoDestroy);
+		AutoDestroyComponent->AttachToComponent(ActorToAutoDestroy->GetRootComponent(), FAttachmentTransformRules::KeepRelativeTransform);
+		AutoDestroyComponent->RegisterComponent();
+	}
+	else
+	{
+		UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("Should only register to auto destroy from the authoritative worker of the actor. Actor %s"), ActorToAutoDestroy->GetDebugName);
```

Comment:
```suggestion
		UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("Should only register to auto destroy from the authoritative worker of the actor. Actor %s"), ActorToAutoDestroy->GetDebugName());
```

Reply:
Changed to *GetNameSafe(ActorToAutoDestroy) to simplify the logging in this function.